### ISSUE TITLE
Mobile-first

### ISSUE THREAD

**Issue by oliverjam** [bobbysebolao/learn-css-grid#2]
We usually try to get them to work mobile-first. Would it be too much to have them start with a one column layout and write their grid to adapt as the viewport gets wider?

I think you could probably get away without media queries at all here by using `auto-fit` but that might get kinda complex.

**Comment by bobbysebolao** [comment]
Yeah, I know what you mean. Although, I thought that starting desktop first for this challenge would be good because it makes the use case for CSS Grid (implementing 2D multi column/row layouts) immediately clear. Whereas, starting off with a one column layout for mobile isn't intuitively something that you need to use Grid for, that could be done with flexbox.

What do you think?

**Comment by oliverjam** [comment]
I mean you don't need to do anything for a one column layout, that's just how the browser lays things out by default.

"mobile-first" in this case would just mean writing their grid code inside a `@media (min-width: 30em)`. The default layout is fine for mobile

**Comment by bobbysebolao** [comment]
I'm not sure I understand. Do you mean that everything inside `style.css` should be wrapped in a `@media (min-width: 30em)` media query?

**Comment by oliverjam** [comment]
Not everything, just any of the grid layout. So you leave everything as `display: block` by default, then for bigger screens where you have space you can add some grid styles to change the layout to make better use of the space.

E.g. in your solution you would put these lines inside the media query (excluding the background image):

https://github.com/bobbysebolao/learn-css-grid/blob/8b26a440c86ef36203a8c42c3eac23724e9c33ff/solution/style.css#L6-L22

**Comment by oliverjam** [comment]
Added this to my example Codepen: https://codepen.io/oliverjam/pen/dyoJGQg?editors=1100
